### DISCUSSION TITLE
Is session table data saved forever, even if it's expired?

### DISCUSSION THREAD

**Discussion by edardev** [ory/kratos#2192] [Q&A] [answered]
I'm wondering what happens after sessions in a Postgres database expire, are they automatically deleted as soon as they expire or will they be stored forever until someone runs a delete operation on them?

**Comment by vinckr** [comment, marked as answer]
AFAIK they will be stored forever at the moment.
There is an open issue on making it an option to clean up the database: https://github.com/ory/kratos/issues/952
